File: Part_2/Python/is_user_in_stance/test_is_user_in_stance_function.py
#!/usr/bin/env python
import os
import matplotlib.pyplot as mplt
from . import is_user_in_stance as iuis
from .. import helper_functions as hf


def test_is_user_in_stance():
	"""Test your code with this function"""
	files = os.listdir("dataset")

	i = 0
	for filename in files:
		data_array = hf.load_csv_file("dataset/" + filename)

		is_user_in_stance_predictions = iuis.is_user_in_stance(data_array)

		mplt.figure(i + 1)
		mplt.plot(data_array[:, hf.variable_name_to_column_index("gait_state")], "b")
		mplt.plot(is_user_in_stance_predictions, "r")
		mplt.draw()

		i += 1

	mplt.show()


if __name__ == "__main__":
	test_is_user_in_stance()

File: Part_2/Matlab/isUserInStance/testIsUserInStanceFunction.m
%% Test your code by running the following
close all;
clear all;
testIsUserInStance();

function testIsUserInStance()
    files = dir('dataset');
    files = files(3:end,1);

    for i = 1:size(files,1)
        filename = files(i).name;
        data_table = readtable(strcat('dataset/',filename));

        isUserInStancePredictions = isUserInStance(data_table);

        figure(i)
        plot(data_table.gait_state, 'b')
        hold on;
        plot(isUserInStancePredictions,'r')
    end
end
